src/ui/actions/event-listeners.ts

```diff
@@ -467,6 +478,16 @@ function createReactEventMapper(eventType: SEARCHABLE_EVENT_TYPES) {
         };
         searchedNodes.push(searchedNode);
 
+        const currentNodeName = currentNode.nodeName.toLowerCase();
+
+        if (
+          injectedValues.eventType === "mousedown" &&
+          currentNodeName === "button" &&
+          (currentNode as HTMLButtonElement).type === "submit"
+        ) {
+          clickWasInsideSubmitButton = true;
```

Comment:
nit: We could just set `res.clickWasInsideSubmitButton` directly here and avoid needing the extra line below:
```js
res.clickWasInsideSubmitButton = clickWasInsideSubmitButton;
```